src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunkReadSide.cs

```diff
@@ -147,11 +150,45 @@ public TFChunkReadSideScavenged(TFChunk chunk, bool optimizeCache, ITransactionF
 			}
 
 			public void Uncache() {
-				_midpoints = null;
+				lock (_lock) {
+					_wantMidpoints = false;
+					_midpoints = null;
+				}
+			}
+
+			public void RequestCaching() {
+				lock (_lock) {
+					_wantMidpoints = true;
+				}
 			}
 
-			public void Cache() {
-				_midpoints = PopulateMidpoints(Chunk.MidpointsDepth);
+			private Midpoint[] GetOrCreateMidPoints(ReaderWorkItem workItem) {
+				// don't use mipoints when reading from memory (why not?)
```

Comment:
> (why not?)
i assume it's because using a cached scavenged chunk is equivalent to having a midpoint created for every posmap entry in that chunk